unit/ChangeValuesTest.cpp

```diff
@@ -99,30 +103,36 @@ TEST_CASE("OnChangeValues call is cropping percentage values",
   ActionListener::RawMessageData msgData;
   msgData.userId = 0;
 
-  ac.SetPercentages(0.1f, 0.0f, 0.0f);
+  ActorValues actorValues;
+  actorValues.healthPercentage = 0.1f;
+  actorValues.magickaPercentage = 0.f;
+  actorValues.staminaPercentage = 0.f;
+  ac.SetPercentages(actorValues);
   auto past = std::chrono::steady_clock::now() - 1s;
   ac.SetLastAttributesPercentagesUpdate(past);
-  p.GetActionListener().OnChangeValues(msgData, 1.0f, 1.0f, 1.0f);
+  actorValues.healthPercentage = 1.f;
+  actorValues.magickaPercentage = 1.f;
+  actorValues.staminaPercentage = 1.f;
+  p.GetActionListener().OnChangeValues(msgData, actorValues);
 
-  auto now = ac.GetLastAttributesPercentagesUpdate();
-  std::chrono::duration<float> timeDuration = now - past;
-  float time = timeDuration.count();
+  std::chrono::duration<float> elapsedTime =
+    std::chrono::steady_clock::now() - past;
 
-  float expectedHealth =
-    baseValues.healRate * baseValues.healRateMult * time / 10000.0f;
-  float expectedMagicka =
-    baseValues.magickaRate * baseValues.magickaRateMult * time / 10000.0f;
-  float expectedStamina =
-    baseValues.staminaRate * baseValues.staminaRateMult * time / 10000.0f;
+  float expectedHealth = baseValues.healRate * baseValues.healRateMult *
+    elapsedTime.count() / 10000.0f;
+  float expectedMagicka = baseValues.magickaRate * baseValues.magickaRateMult *
+    elapsedTime.count() / 10000.0f;
+  float expectedStamina = baseValues.staminaRate * baseValues.staminaRateMult *
+    elapsedTime.count() / 10000.0f;
 
   auto changeForm = ac.GetChangeForm();
 
-  REQUIRE_THAT(changeForm.healthPercentage,
-               Catch::Matchers::WithinAbs(expectedHealth + 0.1f, 0.000001f));
-  REQUIRE_THAT(changeForm.magickaPercentage,
-               Catch::Matchers::WithinAbs(expectedMagicka, 0.000001f));
-  REQUIRE_THAT(changeForm.staminaPercentage,
-               Catch::Matchers::WithinAbs(expectedStamina, 0.000001f));
+  REQUIRE(IsNearlyEqual(expectedHealth + 0.1f,
+                        changeForm.actorValues.healthPercentage));
+  REQUIRE(
+    IsNearlyEqual(expectedMagicka, changeForm.actorValues.magickaPercentage));
+  REQUIRE(
+    IsNearlyEqual(expectedStamina, changeForm.actorValues.staminaPercentage));
```

Comment:
Not critical, but a matcher, which you had previously, would print the exact number ig and this would probably not